async-query/src/test/java/org/opensearch/sql/spark/scheduler/parser/IntervalScheduleParserTest.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.scheduler.parser;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.jobscheduler.spi.schedule.IntervalSchedule;
+import org.opensearch.jobscheduler.spi.schedule.Schedule;
+
+public class IntervalScheduleParserTest {
+
+  private Instant startTime;
+
+  @BeforeEach
+  public void setup() {
+    startTime = Instant.now();
+  }
+
+  @Test
+  public void testParseValidScheduleString() {
+    String scheduleStr = "5 minutes";
+    Schedule schedule = IntervalScheduleParser.parse(scheduleStr, startTime);
+
+    assertEquals(new IntervalSchedule(startTime, 5, ChronoUnit.MINUTES), schedule);
+  }
+
+  @Test
+  public void testParseValidScheduleStringWithDifferentUnits() {
+    String scheduleStr = "2 hours";
+    Schedule schedule = IntervalScheduleParser.parse(scheduleStr, startTime);
+
+    assertEquals(new IntervalSchedule(startTime, 120, ChronoUnit.MINUTES), schedule);
+
+    scheduleStr = "1 day";
+    schedule = IntervalScheduleParser.parse(scheduleStr, startTime);
+
+    assertEquals(new IntervalSchedule(startTime, 1440, ChronoUnit.MINUTES), schedule);
+
+    scheduleStr = "3 weeks";
+    schedule = IntervalScheduleParser.parse(scheduleStr, startTime);
+
+    assertEquals(new IntervalSchedule(startTime, 30240, ChronoUnit.MINUTES), schedule);
+  }
+
+  @Test
+  public void testParseNullSchedule() {
+    Schedule schedule = IntervalScheduleParser.parse(null, startTime);
+    assertNull(schedule);
+  }
+
+  @Test
+  public void testParseScheduleObject() {
+    IntervalSchedule expectedSchedule = new IntervalSchedule(startTime, 10, ChronoUnit.MINUTES);
+    Schedule schedule = IntervalScheduleParser.parse(expectedSchedule, startTime);
+
+    assertEquals(expectedSchedule, schedule);
+  }
```

Comment:
Does this overwrap with 1st test case?

Reply:
the parse interface accept Scheduler object as well, the first one is a string